yarn-project/end-to-end/src/integration_l1_publisher.test.ts

```diff
@@ -371,22 +395,44 @@ describe('L1Publisher integration', () => {
       const expectedData = encodeFunctionData({
         abi: RollupAbi,
         functionName: 'process',
-        args: [`0x${l2Proof.toString('hex')}`, `0x${block.toBufferWithLogs().toString('hex')}`],
+        args: [
+          `0x${block.header.toBuffer().toString('hex')}`,
+          `0x${block.archive.toBuffer().toString('hex')}`,
+          `0x${block.bodyToBuffer().toString('hex')}`,
+          `0x${l2Proof.toString('hex')}`,
+        ],
       });
       expect(ethTx.input).toEqual(expectedData);
 
-      const decoderArgs = [`0x${block.toBufferWithLogs().toString('hex')}`] as const;
-      const decodedHashes = await decoderHelper.read.computeDiffRootAndMessagesHash(decoderArgs);
-      const decodedRes = await decoderHelper.read.decode(decoderArgs);
-      const stateInRollup = await rollup.read.rollupStateHash();
+      const headerDecoderArgs = [`0x${block.header.toBuffer().toString('hex')}`] as const;
+      const decodedHeader = await headerDecoderHelper.read.decode(headerDecoderArgs);
```

Comment:
Ideally we should be checking every single value of the header as mentioned in the sol test to ensure that the decoder works as intended.

Reply:
Now properly testing it all in Decoder.t.sol but do you think it makes sense to test it here? Feels a bit redundant to test the same things in Solidity and TS.

Reply:
Ye think I would be checking it in the sol and not in here. Such that to layout can be made in TS easily, then create the new json and then you can update the solidity and have it check everything.